src/db.ts

```diff
@@ -134,21 +134,20 @@ export class Db {
   public static SYSTEM_JS_COLLECTION = CONSTANTS.SYSTEM_JS_COLLECTION;
 
   /**
-   * Creates a new Db instance
+   * Creates a new Db instance.
+   *
+   * Database name validation occurs at operation time.
    *
    * @param client - The MongoClient for the database.
    * @param databaseName - The name of the database this instance represents.
-   * @param options - Optional settings for Db construction
+   * @param options - Optional settings for Db construction.
    */
   constructor(client: MongoClient, databaseName: string, options?: DbOptions) {
     options = options ?? {};
 
     // Filter the options
     options = filterOptions(options, DB_OPTIONS_ALLOW_LIST);
 
-    // Ensure we have a valid db name
-    validateDatabaseName(databaseName);
```

Comment:
From slack db names should still prevent dots to avoid ambiguity

If we permit dots in the db name, code like this:
```ts
await client.db('aa.bb').command({ create: 'cc.dd.' })
```

Will create a collection named `"cc.dd."` in a database named `"aa"` (the `bb` section is removed silently)

Reply:
I added these changes in, and I editted the PR description accordingly. I'm wondering if this is something we also want to note in the Release Highlight, and if so, how?

Reply:
Release highlight LGTM